### PR TITLE
Editor: intercept corrupt app config case to avoid getting stuck on load

### LINKED ISSUE #1457: Editor fails to launch if the user prefs file is corrupted
**Problem**

This has been reported few times already (e.g. [forum thread](https://www.adventuregamestudio.co.uk/forums/index.php?topic=59358.0)), but if the preferences file gets corrupted for whatever reason Editor refuses to start, displaying "unhandled exception".

**Expected behavior**

Leaving the corruption problem aside, there has to be a way to launch the program even if the prefs cannot be loaded, using defaults.

**To reproduce**

1. Open current preferences file (should be in `%USERPROFILE%\AppData\Local\AGS\AGSEditor.exe_StrongName_mqpdv5kqypl2dezfmjnjxse4zvfvse2h\<VERSION NUMBER>\user.config`) with a text editor.
2. Put random garbage inside and save.
3. Try launching AGS Editor. Observe the exception.

### PR DESCRIPTION
**Pull request by ivan-mogilko**
Resolves #1457

If the application settings file is corrupt Editor threw exception at startup and got stuck at the splash screen.

Added a test in AppSettings constructor that catches any config exceptions and suggests that it should be reset to continue. As a second option it lets close the Editor killing the process (perhaps, may be useful if it was user who was editing the file and left it in an invalid state).

NOTE: I had to add System.Configuration reference to the project.

### CODE REVIEW

**Comment by ericoporto** [comment]
I think it's better to delete and inform instead of offering the option of closing the Editor. People rarely read text pop-up messages, so if they click cancel without reading it will simply appear the editor closed for no reason. I understand manually editing the game project, but the config appears to be something someone would not usually do.

**Comment by morganwillcock** [comment]
I'm not sure it is a good idea to have the check within the constructor of the derived class itself, there might be a risk that a change to the parent class would cause the file to be loaded earlier.

**Comment by ivan-mogilko** [comment]
Variant 2:
* test for corruption in the AGSEditor's constructor; I was not certain how soon the Settings are used, so constructor is probably the safest time.
* Do not ask for confirmation, just report the problem and reset file.

File: Editor/AGS.Editor/AGSEditor.cs [hunk 1]
```diff
@@ -324,6 +326,28 @@ public void Dispose()
...
+            // Test for a valid config file, in case of corrupt config
+            // delete the file and notify the user that the prefs will be reset.
+            try
+            {
+                ConfigurationManager.OpenExeConfiguration(ConfigurationUserLevel.PerUserRoamingAndLocal);
+            }
+            catch (ConfigurationErrorsException ex)
+            {
+                string filename = ex.Filename;
+                MessageBox.Show("Editor's configuration is corrupt and cannot be loaded. " +
```
Comment (persn) [inline Editor/AGS.Editor/AGSEditor.cs:340]: Use `Factory.GUIController.ShowMessage` instead, it'll play nicer when we finally make improvements to editor running in headless mode as discussed in #666

Comment (ivan-mogilko) [reply Editor/AGS.Editor/AGSEditor.cs]: Fixed.

**Comment by ericoporto** [comment]
I tested by going to mine `C:\Users\user\AppData\Local\AGS\AGSEditor.exe_StrongName_mqpdv5kqypl2dezfmjnjxse4zvfvse2h\3.6.0.13` dir and messing up the user.config, and it showed the message box saying it was going to delete and keep going and it deleted and kept going! When I tried the same with the previous version it got stuck and required using the task manager to kill the AGS process, so I say this PR is working!

**Pull request merged**